packages/transaction-controller/src/helpers/GasFeePoller.ts

```diff
@@ -420,3 +421,58 @@ export function updateTransactionGasFees({
     txMeta.layer1GasFee = layer1GasFee;
   }
 }
+
+/**
+ * Updates `txParams` gas values accordingly with given `userFeeLevel` from `txMeta.gasFeeEstimates`.
+ *
+ * @param args - Argument bag.
+ * @param args.txMeta - The transaction meta.
+ * @param args.userFeeLevel - The user fee level.
+ */
+export function applyTransactionGasEstimatesToTransaction({
```

Comment:
Is this not a duplicate of the above function? Could we call this function from the other to remove that duplication?